### PR TITLE
Implement `Automationinterface` class

### LINKED ISSUE #1: Implement AutomationInterface class
- As the class is constructed, it attempts to create a DTE instance. When it destruct, the DTE instance must be closed if necessary
- It will, in the future, have commands for interacting with the automation interface. But for this issue, the scope is to implement the constructor and destructor

### PR DESCRIPTION
**Pull request by ahuca**

### CODE REVIEW

File: src/TwinGet.Cli/TwinGet.Cli.csproj [hunk 1]
```diff
@@ -6,9 +6,10 @@
     <AssemblyTitle>TwinGet Command Line</AssemblyTitle>
     <Description>TwinGet Command Line Interface</Description>
     <OutputType>Exe</OutputType>
-    <TargetFramework>net7.0</TargetFramework>
+    <TargetFramework>net7.0-windows</TargetFramework>
```
Comment (Colho) [inline src/TwinGet.Cli/TwinGet.Cli.csproj:9]: Is the windows framework enforced by something?

File: src/TwinGet.AutomationInterface/AutomationInterface.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                if (t is null) { continue; }
+
+                EnvDTE80.DTE2? dte;
+                try
+                {
+                    dte = (EnvDTE80.DTE2?)Activator.CreateInstance(t);
+                }
+                catch { continue; }
+
+                if (dte is null)
```
Comment (Colho) [inline src/TwinGet.AutomationInterface/AutomationInterface.cs]: to simplify continue usage, could move this if directly under the dte creation and inline it?
if (dte is null) { continue; }

File: src/TwinGet.AutomationInterface/AutomationInterface.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        public string ProgId { get => _progId; }
+        private string _progId;
+        private readonly EnvDTE80.DTE2? _dte;
+        private bool _disposedValue;
+
+        public AutomationInterface()
+        {
+            MessageFilter.Register();
+            _dte = TryInitializeDte(out _progId);
+            if (_dte is null) { throw new CouldNotCreateTwinCatDte("Is TwinCAT installed in this system?"); }
```
Comment (Colho) [inline src/TwinGet.AutomationInterface/AutomationInterface.cs]: Depending on how this message is used, would be better to give more context, that the DTE creation failed, and as an additive tip to make sure TwinCAT is installed.

**Review by Colho — APPROVED** [review]
Can be merged, minor nitpicks and questions

File: src/TwinGet.Cli/TwinGet.Cli.csproj (hunk 1 above)
Comment (ahuca) [reply src/TwinGet.Cli/TwinGet.Cli.csproj:9]: Beckhoff's Automation Interface can only be supported by Windows, so this whole project is limited to Windows platform.

**Pull request merged**